sklearn/model_selection/_search.py

```diff
@@ -999,3 +1004,356 @@ def fit(self, X, y=None, labels=None):
+        explosion of memory consumption when more jobs get dispatched
+        than CPUs can process. This parameter can be:
+
+            - None, in which case all the jobs are immediately
+              created and spawned. Use this for lightweight and
+              fast-running jobs, to avoid delays due to on-demand
+              spawning of the jobs
+
+            - An int, giving the exact number of total jobs that are
+              spawned
+
+            - A string, giving an expression as a function of n_jobs,
+              as in '2*n_jobs'
+
+    iid : boolean, default=True
+        If True, the data is assumed to be identically distributed across
+        the folds, and the loss minimized is the total loss per sample,
+        and not the mean loss across the folds.
+
+    cv : int, cross-validation generator or an iterable, optional
+        Determines the cross-validation splitting strategy.
+        Possible inputs for cv are:
+          - None, to use the default 3-fold cross-validation,
+          - integer, to specify the number of folds.
+          - An object to be used as a cross-validation generator.
+          - An iterable yielding train/test splits.
+
+        For integer/None inputs, if ``y`` is binary or multiclass,
+        :class:`StratifiedKFold` used. If the estimator is a classifier
+        or if ``y`` is neither binary nor multiclass, :class:`KFold` is used.
+
+        Refer :ref:`User Guide <cross_validation>` for the various
+        cross-validation strategies that can be used here.
+
+    refit : boolean, default=True
+        Refit the best estimator with the entire dataset.
+        If "False", it is impossible to make predictions using
+        this RandomizedSearchCV instance after fitting.
+
+    verbose : integer
+        Controls the verbosity: the higher, the more messages.
+
+    random_state : int or RandomState
+        Pseudo random number generator state used for random uniform sampling
+        from lists of possible values instead of scipy.stats distributions.
+
+    error_score : 'raise' (default) or numeric
+        Value to assign to the score if an error occurs in estimator fitting.
+        If set to 'raise', the error is raised. If a numeric value is given,
+        FitFailedWarning is raised. This parameter does not affect the refit
+        step, which will always raise the error.
+
+    n_candidates : int, default=500
+        Number of random candidates to sample for each GP iterations
+        Default is 500.
+
+    gp_params : dict, default={}
+        Parameters to pass to the GaussianProcessRegressor object.
+
+    xi : float, default=0.01
+        Exploration-exploitation trade-off parameter for the expected
+        improvement acquisition function. Higher values for this parameter
+        yield updates that are more conservative in which the exploitation
+        criterion dominates over the exploration criterion.
+
+    kappa: float, default=1.96
```

Comment:
We could also support an automatic schedule for decreasing kappa over time (kappa="auto"), for instance as defined in  [ Dynamic Learning Gaussian Process Bandits](http://www.cs.ucl.ac.uk/fileadmin/UCL-CS/research/RN_12_15.pdf)  for beta_t at the top of page 4 (this one is derived from bounding the regret).